tests/Core/Tokenizer/EnumTest.inc

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/* testPureEnum */
+enum Foo
+{
+    case SOME_CASE;
+}
+
+/* testBackedIntEnum */
+enum Boo: int {
+    case ONE = 1;
+    case TWO = 1;
+}
+
+/* testBackedStringEnum */
+enum Hoo: string
+{
+    case ONE = 'one';
+    case TWO = 'two';
+}
+
+/* testComplexEnum */
+enum ComplexEnum: int implements SomeInterface
+{
+    use SomeTrait {
+        traitMethod as enumMethod;
+    }
+
+    const SOME_CONSTANT = true;
+
+    case ONE = 1;
+
+    public function someMethod(): bool
+    {
+    }
+}
+
+/* testEnumUsedAsClassName */
+class Enum {
+    /* testEnumUsedAsClassConstantName */
+    const ENUM = 'enum';
+
+    /* testEnumUsedAsMethodName */
+    public function enum() {
+        // Do something.
+
+        /* testEnumUsedAsPropertyName */
+        $this->enum = 'foo';
+    }
+}
+
+/* testEnumUsedAsFunctionName */
+function enum()
+{
+}
```

Comment:
Could tests be added for the following ?

```php
/* testDeclarationContainingComment */
enum /* comment */ Name
{
    case SOME_CASE;
}

/* testParseErrorMissingname */
enum {
    case SOME_CASE;
}

/* testParseErrorLiveCoding */
// This must be the last test in the file.
enum
```

I also wonder about the below, not sure if it _could_ be valid, but would be interesting to ensure the tokenization across PHP versions is consistent:
```php
enum Enum {}
```

And for the non-enum test cases - what about adding these ? (note: this is more to safeguard that these are tokenized as `T_STRING` across all supported PHP versions than to check the `T_ENUM` backfill):
```php
namespace My\Enum\Collection;
$obj = new Enum;
$var = enum($a, $b);
var = namespace\enum();
echo Enum::CONSTANT;
echo ClassName::ENUM;
// etc
```

Reply:
Thanks, added.